File: UCSample/UCSample/NetParamsController.m
//
//  NetParamsController.m
//  UCSample
//
//  Copyright (c) 2016 PortSIP Solutions, Inc. All rights reserved.
//

#import "NetParamsController.h"

@interface NetParamsController ()

@end

@implementation NetParamsController

- (void)viewDidLoad {
  [super viewDidLoad];
  [self.navigationController setNavigationBarHidden:NO];
  [_tbView setDataSource:self];
  [_tbView setDelegate:self];
}

- (IBAction)backButtonClicked:(id)sender {
  [self.navigationController popToRootViewControllerAnimated:YES];
}

- (NSInteger)tableView:(UITableView *)tableView
    numberOfRowsInSection:(NSInteger)section {
  if (self.data) {
    return self.data.count;
  }
  return 0;
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
  if (!self.data || self.data.count == 0) {
    return nil;
  }

  UITableViewCell *cell =
      [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
                             reuseIdentifier:self.data[indexPath.row]];
  cell.textLabel.text = self.data[indexPath.row];
  return cell;
}

- (CGFloat)itemCellHeight:(NSIndexPath *)indexPath {
  return 44.0f;
}

- (void)tableView:(UITableView *)tableView
    didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
  [self.navigationController popToRootViewControllerAnimated:YES];
  [self.delegate didSelectValue:self.title value:indexPath.row];
}
@end

File: UCSample_swift/UCSample/NetParamsController.swift
//
//  NetParamsController.swift
//  UCSample
//
//  Created by portsip on 16/6/30.
//  Copyright © 2016 PortSIP. All rights reserved.
//
import UIKit
protocol NetParamsControllerDelegate {
    func didSelectValue(_ key: String, value: Int)

//    func didEndSelectValue(key:String,value:Int)
}

class NetParamsController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    @IBOutlet var tbView: UITableView!
    var data: [String]! = []
    var labletitle: String!
    var delegate: AnyObject!

    override func viewDidLoad() {
        super.viewDidLoad()
        navigationController!.setNavigationBarHidden(false, animated: false)
        // self.navigationController!.title = labletitle;
        tbView.dataSource = self
        tbView.delegate = self
    }

    @IBAction func backButtonClicked(_: AnyObject) {
        navigationController!.popToRootViewController(animated: true)
    }

    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
        if data != nil {
            return data.count
        }
        return 0
    }

    func tableView(_: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: UITableViewCell.CellStyle.default, reuseIdentifier: data[(indexPath as NSIndexPath).row])
        cell.textLabel!.text = data[(indexPath as NSIndexPath).row]
        return cell
    }

    func itemCellHeight(_: IndexPath) -> Float {
        44.0
    }

    func tableView(_: UITableView, didSelectRowAt indexPath: IndexPath) {
        navigationController!.popToRootViewController(animated: true)
        let tempdelegate = delegate as! NetParamsControllerDelegate
        tempdelegate.didSelectValue(labletitle, value: (indexPath as NSIndexPath).row)
    }
}
